src/main/java/org/openstreetmap/atlas/checks/validation/areas/WaterbodyAndIslandSizeCheck.java

```diff
@@ -0,0 +1,336 @@
+package org.openstreetmap.atlas.checks.validation.areas;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.atlas.predicates.TagPredicates;
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.Area;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.geography.atlas.items.RelationMember;
+import org.openstreetmap.atlas.tags.NaturalTag;
+import org.openstreetmap.atlas.tags.PlaceTag;
+import org.openstreetmap.atlas.tags.RelationTypeTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * This check identifies waterbodies and islands that are either too small, or too large in size.
+ * Minimum surface area values are measured in meters squared; while maximum values are measured in
+ * kilometers squared. An island can either be tagged as place=island, place=islet,
+ * place=archipelago, or a MultiPolygon Relation with the type=multipolygon and natural=water tags.
+ *
+ * @author danielbaah
+ */
+public class WaterbodyAndIslandSizeCheck extends BaseCheck
+{
+
+    private static final long serialVersionUID = 1L;
+    private static String WATERBODY_MIN_INSTRUCTION = "Waterbody {0,number,#} has an area {1,number,#.##} m^2 which is too small.";
+    private static String WATERBODY_MAX_INSTRUCTION = "Waterbody {0,number,#} has an area {1,number,#.##} km^2 which is too large.";
+    private static String ISLAND_MIN_INSTRUCTION = "Island {0,number,#} has an area {1,number,#.##} m^2 which is too small.";
+    private static String ISLAND_MAX_INSTRUCTION = "Island {0,number,#} has an area {1,number,#.##} km^2 which is too large.";
+    private static String ISLET_MIN_INSTRUCTION = "Islet {0,number,#} has an area {1,number,#.##} m^2 which is too small.";
+    private static String ISLET_MAX_INSTRUCTION = "Islet {0,number,#} has an area > 1 km^2 and should likely be tagged with place=ISLAND.";
+
+    private static final Predicate<AtlasObject> IS_ISLET = object -> Validators.isOfType(object,
```

Comment:
Let's move the predicates below the defaults to keep all constants/variables together.